add early draft for ecosystem overview

Vindaar opened the pull request:
Obviously a super early draft. Had some time right now though and started to write a list of things that came to my mind. I probably miss a good amount of stuff here.

HugoGranstrom commented on book/overview/index.nim:
```diff
@@ -0,0 +1,170 @@
+## Flambeau
+
+[Flambeau](https://github.com/scinim/flambeau) is a [libtorch](https://pytorch.org/) wrapper. Thus,
+it provides both a general `Tensor[T]` type (and the expected associated operations) as well as
```
I may be splitting hairs now, but calling `flambeau` "general" sounds odd when you compare it to Arraymancer which is super-general in comparison. But sure for numeric datatypes it is general. Maybe using "numerical" instead of general?

Vindaar replied:
haha, yeah. The "general" here was just meant as a comparison to the `Tensor[T]` in arraymancer. Not the possible operations on it. But you're right, it sounds confusing. All the text is, uhh, to be properly written anyway.

HugoGranstrom replied:
I found the text that was written quite good and concise otherwise :)

HugoGranstrom commented on book/overview/index.nim:
```diff
@@ -0,0 +1,170 @@
+it provides both a general `Tensor[T]` type (and the expected associated operations) as well as
+being a machine learning library.
+
+## What am I
```
Missing a "missing"? ;)

Vindaar replied:
No, it's referring to the fact that I don't know where the packages in the section belong. Hence a "What am I", asks the package ;)

HugoGranstrom replied:
Ahhhh :D

HugoGranstrom reviewed (commented):
Just has a few small silly things to note. Nothing missing that I can think of right now at least :)

Vindaar commented:
@brentp Could you maybe tell me which (or all?) of your libraries should be included here? And maybe with a couple of words about them? 

@SciNim/sciadmins If anyone else has any input, just let me know!

HugoGranstrom commented:
Looking good to me, discovered a lot of new libraries I've never seen before :D

pietroppeter commented:
this is very nice and much welcome!

as a first remark:
- GSL is listed in others but it could be listed in many different places (special functions, matrices, numerical algorithms, statistics, fft, ...), see functional areas: https://www.gnu.org/software/gsl/

Some time ago I started compiling a star list (a rather recent GitHub feature) dedicated to scinim libraries: https://github.com/stars/pietroppeter/lists/scinim

Do the libraries that appear here should have been tested by someone other than the author? I have not yet the opportunity to test many of them but here are some from that list that might be worth adding to this overview.

- https://github.com/ayman-albaz/special-functions (along with c-blake spfun could make a Special functions section where GSL should probably also appear)

a new statistics section could be made with these:

- https://github.com/ayman-albaz/statistical-tests
- https://github.com/ayman-albaz/linear-models
- https://github.com/ayman-albaz/distributions

GSL should also go in this statistics section and probably also alea (and we could call it Probability/Statistics).

As a note for alea:
- I would probably not call alea a library for random number generators (which is usually the term reserved for the primitive RNG) but more for managing random variables according to various distributions. It could be file under a Probability section along with ayman-albaz's distributions. Note also that alea depends on a RNG library which could also be added (https://github.com/oprypin/nim-random); not sure also why alea depends on that rather than the stdlib generator (maybe at the time it was missing some functionalities.

Another library from @ayman-albaz (let me tag him, he might have other suggestions) not sure where we could put this (other?): https://github.com/ayman-albaz/distances

And this might not be a "fundamental" data handling but it could be handy: https://github.com/ayman-albaz/nim-duckdb

In matrices we could add this: https://github.com/planetis-m/manu

I am not sure if they should be part of this overview but stuff like inimitable, jupyternim and nimib could be cited in this context.

A new section algebra could be made with:

- https://github.com/MichalMarsalek/nim-algebra
- https://github.com/andreaferretti/emmy

there are few others in that list (for NLP, deeplearning, ...) but for the moment I will stop here. :)

brentp commented:
Hi, @Vindaar ,
[hts-nim](https://github.com/brentp/hts-nim) is probably the main one. It's used at a couple of companies, including PacBio. It is also used by a number of my published tools, including [mosdepth](https://github.com/brentp/mosdepth) which has almost 250 citations. Most of my stuff is command-line tools, so less appropriate for this list.

ayman-albaz commented:
@pietroppeter 
All the libraries you linked from me can be a part of SciNim. I think a good idea is that anything that falls under python's SciPy could be "Statistics". That would include the libraries you linked.

While DuckDB is just a database like sqlite, it is a database that focus more on fast-analysis rather than fast-transactions (sqlite). Perhaps you could create a section called "Tools" or "I/O", and add it there? Additional things that can be put in the "Tools" section are perhaps: https://github.com/emef/nimarrow_glib and https://github.com/Vindaar/nimhdf5. Basically anything related to I/O related to scientific programming.

mratsim commented:
> Additional things that can be put in the "Tools" section are perhaps: https://github.com/emef/nimarrow_glib and https://github.com/Vindaar/nimhdf5. Basically anything related to I/O related to scientific programming.

Proper name would be serialization formats

Vindaar commented:
Just gave this a big update. Let's see what it looks like once the preview is ready.

Vindaar commented:
Removed the long format stuff at the beginning (let's keep this an overview and at some point add some 'recipe' like thing elsewhere). Also added `nimib`, `inim` and `nim-sos`.

If this looks fine and no one has any other ideas, I'll merge this in the next few days.

pietroppeter commented:
It looks very nice and I look forward to try all the goodies listed there :)

A couple of very minor remarks:
- given that you have provided short description for almost all of the libraries, you could add them for the few remainings without short description (statistical tests, distributions, linear models; weave, taskpools; in principle also bridges py R julia and cuda/lapack/blas/cl wrappers...)
- clustering is mentioned in the distances section but it probably might make more sense to mention it in the ML section (or in both)

Vindaar commented:
> on Nim Italia telegram channel someone was looking for Linear Programming stuff and I think there is nothing for Nim, right? this reminded though that we could add https://github.com/zevv/nimz3 to this list.

Good idea! Will add it (and apply the other changes you mentioned).

> Also I am realizing now this could well be a markdown file since there is no special logic and CI would be faster.

Haha, good point. Initially there was supposed to be code of course. How do I include a pure markdown file?

pietroppeter commented:
> How do I include a pure markdown file?

just change the extension in the toc to .md (we probably could automate detection also of md extnesion) and replace the nim file with the md file

Vindaar commented:
If the preview builds successfully I'll merge this later.
